scene/gui/popup_menu.cpp

```diff
@@ -106,6 +107,14 @@ Size2 PopupMenu::_get_contents_minimum_size() const {
 	return minsize;
 }
 
+void PopupMenu::set_extra_height_limit(int p_limit) {
+	extra_height_limit = p_limit;
+}
+
+int PopupMenu::get_extra_height_limit() const {
+	return extra_height_limit;
+}
+
```

Comment:
I don't really like the name "set extra height limit". For someone coming to this code in 1 year's time, they would need to look into where the `extra_height_limit` property was used to understand what it actually did.

"Height Limit" implies that if it was set to say, 50, the maximum height of the popup would be 50. But that is not what this does. This `extra_height_limit` is _subtracted_ from the actual height limit (usable parent rect). Essentially what it does is set a maximum Y position for the popup menu.

As a result, perhaps changing the name to something like `set_highest_point(int p_value)` would represent it better. I still don't think that `set_highest_point` is necessarily the best name, but I just want to get the idea of a name change out there.

Reply:
Agree.